### PR TITLE
Additional Port Interface Types

### PR DESCRIPTION
**Pull request by phshaikh**
Summary:

Providing more options for port interface types to allow more granular programmability of port settings. These interface types are part of  IEEE 802.3 standard

GMII - Gigabit
SFI - 10 Gigabit
XLAUI - 40 Gigabit
CAUI - 100 Gigabit

Each of these have different electrical characteristics and different from available port interface types.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: inc/saiport.h [hunk 1]
```diff
@@ -236,6 +236,24 @@ typedef enum _sai_port_interface_type_t
...
+    SAI_PORT_INTERFACE_TYPE_SFI,
+
+    /** Interface type XLAUI */
+    SAI_PORT_INTERFACE_TYPE_XLAUI,
+
+    /** Interface type CUSTOM BEGIN */
+    SAI_PORT_INTERFACE_TYPE_CUSTOM_BEGIN,
+
+    /** Interface type CUSTOM END */
+    SAI_PORT_INTERFACE_TYPE_CUSTOM_END = SAI_PORT_INTERFACE_TYPE_CUSTOM_BEGIN,
```
Comment (kcudnik) [inline inc/saiport.h:256]: Can you change that to: SAI_PORT_INTERFACE_TYPE_CUSTOM_BASE =  0x10000000
this way it will follow scheme like in other enums

Comment (phshaikh) [reply inc/saiport.h:256]: Yes, i will do it shortly!

Comment (phshaikh) [reply inc/saiport.h]: @kcudnik done!

File: inc/saiport.h [hunk 2]
```diff
@@ -236,6 +236,24 @@ typedef enum _sai_port_interface_type_t
...
+
+    /** Interface type GMII */
+    SAI_PORT_INTERFACE_TYPE_GMII,
+
+    /** Interface type SFI */
+    SAI_PORT_INTERFACE_TYPE_SFI,
+
+    /** Interface type XLAUI */
+    SAI_PORT_INTERFACE_TYPE_XLAUI,
+
```
Comment (JaiOCP) [inline inc/saiport.h:251]: @kcudnik Do we need to add SAI_PORT_INTERFACE_TYPE_MAX or extensions can be added using CUSTOM_RANGE_START.

Comment (phshaikh) [reply inc/saiport.h:251]: how do we plan to use SAI_PORT_INTERFACE_TYPE_MAX?

**Pull request merged**